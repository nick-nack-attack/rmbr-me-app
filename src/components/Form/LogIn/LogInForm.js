import React, { Component, useState, useContext } from 'react';
import AuthApiService from '../../../services/auth-api-service';
import RmbrmeContext from "../../../contexts/RmbrmeContext";
import { Button, Input } from '../../Utils/Utils';

import "../Form.scss";
import Loader from '../../Utils/Loader';

const LogIn = props => {

    const [ email, setemail ] = useState();
    const [ password, setPassword ] = useState();
    const [ buttonLabel, setButtonLabel ] = useState('Welcome back!');
    const [ error, seterror ] = useState(null);

    let context = useContext(RmbrmeContext);

    let handleSubmitJwtAuth = ev => {
        setButtonLabel('Submitting Login...');
        ev.preventDefault();
        seterror(null);
        let userLogin = {
            user_name: email,
            password: password
        };
        AuthApiService.postLogin(userLogin)
            .then(() => {
                setPassword('');
                setemail('');
                props.onLoginSuccess();
            })
            .catch(res => {
                setButtonLabel('Somthing went wrong');
                seterror(res.error);
            })
    };

    let handleDemoLogin = ev => {
        setButtonLabel('Submitting Login...');
        ev.preventDefault();
        seterror(null);

        let userLogin = {
            user_name: 'JoeBang@gmail.com',
            password: '$uperS1pper'
        };

        AuthApiService.postLogin(userLogin)
            .then(() => {
                this.setState({
                    user_name: '',
                    password: ''
                })
                props.onLoginSuccess();
            })
            .catch(res => {
                setButtonLabel('Somthing went wrong');
                seterror(res.error);
            })
    };

    return (
        <>
            <form
                className='LoginForm'
                onSubmit={e => handleSubmitJwtAuth(e)}
            >
                <legend>
                    <h2>
                        Log in {}
                    </h2>
                </legend>
                <div
                    className='user_name'
                >
                    <label
                        htmlFor='LoginForm__user_name'
                    >
                        Email
                    </label>
                    <Input
                        required
                        name='user_name'
                        type='email'
                        id='LoginForm__user_name'
                        value={email}
                        onChange={e => setemail(e.target.value)}
                    >
                    </Input>
                </div>
                <div
                    className='password'
                >
                    <label
                        htmlFor='LoginForm__password'
                    >
                        Password
                    </label>
                    <Input
                        required
                        name='password'
                        type='password'
                        id='LoginForm__password'
                        onChange={e => setPassword(e.target.value)}
                    >
                    </Input>
                </div>
                <Button
                    label="Login"
                    type="submit"
                />
                <div
                    role='alert'
                >
                    { 
                        error 
                            ? <p className='error_message'>{error.message || error}</p> 
                            : <p className="info_message">{ buttonLabel }</p>}
                </div>
            </form>
            <Button
                label="Login with Demo Account"
                className='login_with_demo_account_button'
                onClick={e => handleDemoLogin(e)}
            />
        </>
    )

}

export default LogIn;