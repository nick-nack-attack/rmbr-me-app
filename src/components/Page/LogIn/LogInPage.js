import React, { Component } from 'react';

// component
import LogInForm from '../../Form/LogIn/LogInForm';

export default class LogInPage extends Component {

    static defaultProps = {
        location: {},
        history: {
            push: () => {},
        }
    };

    handleLoginSuccess = () => {
        window.location.reload(true);
    };

    render() {
        return (
                <LogInForm
                    onLoginSuccess={(ev) => this.handleLoginSuccess(ev)}
                />
        );
    };
};