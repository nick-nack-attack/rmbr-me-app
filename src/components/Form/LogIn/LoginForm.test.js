import React from 'react';
import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'
import LogIn from "./LogIn";

describe(`LogIn form component`, () => {

    it('renders LogInForm by default', () => {
        const wrapper = shallow(<LogInForm/>)
        expect(toJson(wrapper)).toMatchSnapshot()
    })

})