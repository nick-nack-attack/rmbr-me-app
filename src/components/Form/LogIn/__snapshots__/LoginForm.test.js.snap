// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LogInForm component renders LogInForm by default 1`] = `
<Fragment>
  <form
    className="LoginForm"
    onSubmit={[Function]}
  >
    <legend>
      <h2>
        Log in
      </h2>
    </legend>
    <div
      className="user_name"
    >
      <label
        htmlFor="LoginForm__user_name"
      >
        Email
      </label>
      <Input
        id="LoginForm__user_name"
        name="user_name"
        onChange={[Function]}
        required={true}
      />
    </div>
    <div
      className="password"
    >
      <label
        htmlFor="LoginForm__password"
      >
        Password
      </label>
      <Input
        id="LoginForm__password"
        name="password"
        onChange={[Function]}
        required={true}
        type="password"
      />
    </div>
    <Button
      type="submit"
    >
      Login
    </Button>
    <div
      role="alert"
    />
  </form>
  <Button
    className="login_with_demo_account_button"
    onClick={[Function]}
  >
    Log in with Demo Account
  </Button>
</Fragment>
`;
